src/main/java/com/aws/greengrass/shadowmanager/sync/model/CloudUpdateSyncRequest.java

```diff
@@ -190,19 +178,14 @@ public void execute(SyncContext context) throws RetryableException, SkipSyncRequ
      */
     @Override
     public boolean isUpdateNecessary(SyncContext context) throws SkipSyncRequestException, UnknownShadowException {
-        // synchronizing access to shadow sync information because it is accessed by multiple threads.
-        // In this method, shadow sync information is accessed within context of an IPC thread.
-        // The other place is the 'execute' method of this class, which runs on the sync thread.
-        synchronized (context.getSynchronizeHelper().getThingShadowLock(this)) {
-            Optional<ShadowDocument> shadowDocument = context.getDao().getShadowThing(getThingName(), getShadowName());
+        Optional<ShadowDocument> shadowDocument = context.getDao().getShadowThing(getThingName(), getShadowName());
```

Comment:
removed the `synchronized` block since not we're not reading the sync info from multiple threads anymore